### PR TITLE
The first minimal test

### PR DESCRIPTION
**Pull request by FXTi**
To solve issue: https://github.com/FXTi/iocp-wrapper/issues/5

* Add GetQueuedCompletionStatusEx()

### CODE REVIEW

**Comment by FXTi** [comment]
@piscisaureus Could you please review this?

**Comment by FXTi** [comment]
Rebased on https://github.com/FXTi/iocp-wrapper/commit/94f7dff30f2fe14f7e2b0199f3906eb480d99cc7

**Comment by carllerche** [comment]
Looks like you got something compiling :+1:. What happens if you run it?

**Comment by FXTi** [comment]
@carllerche
If change the timeout value to be 0. It would log this:



Since this is still a minimal test, this should prove the code is correct and does poll with afd driver and SIO handle. Although I would wait for confirm from @piscisaureus.

And I add conversion step as preparation to test events relates functions, the conversion will eventually refactored in Mio's style.

File: src/main.rs [hunk 1]
```diff
@@ -246,5 +312,40 @@ fn main() {
...
+        println!("    Internal: {:?}", ele.Internal);
+        println!(
+            "    dwNumberOfBytesTransferred: {:?}",
+            ele.dwNumberOfBytesTransferred
+        );
+    }
+
+    //WSACleanup is used to terminate the use of the WS2_32 DLL.
+    unsafe {
+        WSACleanup();
```
Comment (piscisaureus) [inline src/main.rs]: This is only (minimally) relevant when building a shared library, not when building an executable.
Also, nobody ever unloads winsock, just like nobody ever "unloads" libc. Don't waste your time with it.

Comment (FXTi) [reply src/main.rs]: Okay, I will remove that.

**Comment by FXTi** [comment]
`afd_poll()` return error code: ERROR_INVALID_HANDLE

**Comment by FXTi** [comment]
Now: https://github.com/FXTi/iocp-wrapper/pull/10/commits/b593b62cf61f4c42ce15583974f5882d2298fbf0

I try to show error code from NtCreateFile() in afd_create_helper_handle().

It's 0x3E6(ERROR_NOACCESS) : Invalid access to memory location.

Suspect it's lazy_static related problem...

**Comment by FXTi** [comment]
Now: https://github.com/FXTi/iocp-wrapper/pull/10/commits/63248bd0012a842b7dd20e8135706dc6e2211eb5

It now can work properly, but events return zero.



Events shows `0x0`

Do the work like wepoll, in https://github.com/FXTi/iocp-wrapper/pull/10/commits/feac8be63f779220254efa46b3dba2057aa24ad5

**Comment by FXTi** [comment]
Now it works!

**Pull request merged**